mypyc/genops.py

```diff
@@ -3408,65 +3452,67 @@ def visit_ellipsis(self, o: EllipsisExpr) -> Value:
         return self.primitive_op(ellipsis_op, [], o.line)
 
     # Unimplemented constructs
-    # TODO: some of these are actually things that should never show up,
-    # so properly sort those out.
-
-    def visit__promote_expr(self, o: PromoteExpr) -> Value:
-        raise NotImplementedError
-
     def visit_await_expr(self, o: AwaitExpr) -> Value:
-        raise NotImplementedError
+        self.bail("await is unimplemented", o.line)
 
-    def visit_backquote_expr(self, o: BackquoteExpr) -> Value:
-        raise NotImplementedError
+    def visit_yield_from_expr(self, o: YieldFromExpr) -> Value:
+        self.bail("yield from is unimplemented", o.line)
 
     def visit_complex_expr(self, o: ComplexExpr) -> Value:
-        raise NotImplementedError
+        self.bail("complex literals are unimplemented", o.line)
 
-    def visit_enum_call_expr(self, o: EnumCallExpr) -> Value:
-        raise NotImplementedError
+    def visit_star_expr(self, o: StarExpr) -> Value:
+        self.bail("Star expressions (in non call contexts) are unimplemented", o.line)
 
-    def visit_exec_stmt(self, o: ExecStmt) -> None:
-        raise NotImplementedError
+    def visit_reveal_expr(self, o: RevealExpr) -> Value:
+        self.bail("reveal_from can't be run", o.line)
```

Comment:
The message is confusing -- it should depend on the kind (`reveal_type` vs. `reveal_locals`).

There is no test case for this.